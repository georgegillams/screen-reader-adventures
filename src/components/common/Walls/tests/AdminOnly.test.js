import React from 'react';
import { render } from '@testing-library/react';

import { AdminOnly } from '..';

describe('<AdminOnly />', () => {
  it('Should render correctly', () => {
    const { container } = render(<AdminOnly user={null}>Test</AdminOnly>);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly with user', () => {
    const { container } = render(<AdminOnly user={{}}>Test</AdminOnly>);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly with admin user', () => {
    const { container } = render(<AdminOnly user={{ admin: true }}>Test</AdminOnly>);

    expect(container).toMatchSnapshot();
  });
});
