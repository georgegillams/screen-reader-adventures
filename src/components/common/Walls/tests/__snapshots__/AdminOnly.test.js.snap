// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminOnly /> Should render correctly 1`] = `
<div>
  <div
    class="section__outer"
  >
    <h2
      class="section__heading"
      id="admin-only"
    >
      Admin only
    </h2>
    <span
      class="paragraph"
    >
      You need to be logged in with an admin account to view this content.
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/login"
      >
        Got a different admin account? Log in here.
      </a>
    </span>
  </div>
</div>
`;

exports[`<AdminOnly /> Should render correctly with admin user 1`] = `
<div>
  Test
</div>
`;

exports[`<AdminOnly /> Should render correctly with user 1`] = `
<div>
  <div
    class="section__outer"
  >
    <h2
      class="section__heading"
      id="admin-only"
    >
      Admin only
    </h2>
    <span
      class="paragraph"
    >
      You need to be logged in with an admin account to view this content.
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/login"
      >
        Got a different admin account? Log in here.
      </a>
    </span>
  </div>
</div>
`;
